lib/wallaby/driver.ex

```diff
@@ -0,0 +1,158 @@
+defmodule Wallaby.Driver do
+  @moduledoc false
+
+  alias Wallaby.{Element, Query, Session}
+
+  @type reason :: any
+  @type url :: String.t
+  @type open_dialog_fn :: ((Session.t) -> any)
+  @type window_dimension :: %{String.t => pos_integer, String.t => pos_integer}
+
+  @doc """
+  Invoked to start a browser session.
+  """
+  @callback start_session(Keyword.t) :: {:ok, Session.t} | {:error, reason}
+
+  @doc """
+  Invoked to stop a browser sesssion.
+  """
+  @callback end_session(Session.t) :: :ok | {:error, reason}
+
+  @doc """
+  Invoked to accept one alert triggered within `open_dialog_fn` and return the alert message.
+  """
+  @callback accept_alert(Session.t, open_dialog_fn) :: {:ok, [String.t]}
+
+  @doc """
+  Invoked to accept one confirm triggered within `open_dialog_fn` and return the confirm message.
+  """
+  @callback accept_confirm(Session.t, open_dialog_fn) :: {:ok, [String.t]}
+
+  @doc """
+  Invoked to accept all open dialogs.
+  """
+  @callback accept_dialogs(Session.t) :: {:ok, String.t}
+
+  @doc """
+  Invoked to accept one prompt triggered within `open_dialog_fn` and return the prompt message.
+  """
+  @callback accept_prompt(Session.t, String.t | nil, open_dialog_fn) :: {:ok, [String.t]}
+
+  @doc """
+  Invoked to retrieve cookies for the given session.
+  """
+  @callback cookies(Session.t) :: {:ok, [%{String.t => String.t}]} | {:error, reason}
+
+  @doc """
+  Invoked to get the current path of the browser's session.
+  """
+  @callback current_path!(Session.t) :: String.t | nil | no_return
```

Comment:
This doesn't need to be fixed in this PR but I think we should remove these in favor of always using `{:ok, result} | {:error, reson}` in the driver modules.